### PR TITLE
add builder

### PR DESCRIPTION
**Pull request by phyzical**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	docker build . --platform=amd64 -t ${DOCKER_ECR}publish_app
+	docker tag ${DOCKER_ECR}publish_app ${DOCKER_ECR}publish_app:latest
+push:
+	docker push ${DOCKER_ECR}publish_app:latest
+login:
+	@if [ "${DOCKER_ECR}" != "" ]; then \
+		aws-vault exec shared_services -- aws ecr get-login-password --region ap-southeast-2 | docker login --username AWS --password-stdin ${DOCKER_ECR}; \
+	fi
+run:
+	docker run ${DOCKER_ECR}publish_app discovery-service asd asd test,production
```
Comment (matthew-puku) [inline Makefile:16]: maybe something not `asd asd` would be more informative

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+.env
```
Comment (matthew-puku) [inline .gitignore:1]: nit: newline, editorconfig

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
+name: Build Docker Image
+on:
+  push:
+
+jobs:
+  push:
+    concurrency:
+      group: ${{ github.event_name }}-${{ github.ref_name == 'master' && github.sha || github.ref_name }}-build
```
Comment (matthew-puku) [inline .github/workflows/build.yml:8]: main?

**Review by matthew-puku — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 3 above)
Comment (matthew-puku) [reply .github/workflows/build.yml:8]: also elsewhere

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      - name: Build Production OCI Image
+        run: |
+          docker pull $IMAGE_ID:latest || true
+          docker build --cache-from $IMAGE_ID:latest \
+          --build-arg DOCKER_ECR=${{ secrets.DOCKER_ECR }} \
+          --build-arg RELEASE_VERSION=${IMAGE_TAG} \
+          -t $IMAGE_ID:$IMAGE_TAG .
+
+      - name: Push to ECR
+        if: github.event_name == 'push' && github.ref_name == 'master'
```
Comment (phyzical) [inline .github/workflows/build.yml:53]:
```suggestion
        if: github.event_name == 'push' && github.ref_name == 'main'
```

File: .gitignore (hunk 2 above)
Comment (phyzical) [inline .gitignore:1]:
```suggestion
.env

```

File: Makefile (hunk 1 above)
Comment (phyzical) [reply Makefile:16]: ah its redundant anyway was testing a way to get it working

Comment (phyzical) [inline Makefile:16]:
```suggestion
```

File: .github/workflows/build.yml (hunk 3 above)
Comment (phyzical) [inline .github/workflows/build.yml:8]:
```suggestion
      group: ${{ github.event_name }}-${{ github.ref_name == 'main' && github.sha || github.ref_name }}-build
```

**Review by matthew-puku — APPROVED** [review]

**Pull request merged**